modules/REST/V2/Languages.php

```diff
@@ -0,0 +1,737 @@
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\REST\V2;
+
+use PLL_Language;
+use PLL_Model;
+use stdClass;
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Allows to access languages via the REST API.
+ *
+ * @since 3.7
+ *
+ * @phpstan-type Fields array{
+ *     code?: non-empty-string,
+ *     locale?: non-empty-string,
+ *     name?: non-empty-string,
+ *     direction?: 'ltr'|'rtl',
+ *     order?: int,
+ *     flag?: string,
+ *     set_default_cat?: bool
+ * }
+ */
+class Languages extends WP_REST_Controller {
+	/**
+	 * @var PLL_Model
+	 */
+	public $model;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param PLL_Model $model Polylang's model.
+	 */
+	public function __construct( PLL_Model $model ) {
+		$this->namespace = 'pll/v2';
+		$this->rest_base = 'languages';
+		$this->model     = $model;
+	}
+
+	/**
+	 * Registers the routes for languages.
+	 *
+	 * @since 3.7
+	 *
+	 * @return void
+	 */
+	public function register_routes(): void {
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}",
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => '__return_true',
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}/(?P<code>[a-z_-]+)",
+			array(
+				'args'   => array(
+					'code' => array(
+						'description' => __( 'Unique code for the language.', 'polylang' ),
+						'type'        => 'string',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => '__return_true',
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves a collection of languages.
```

Comment:
As of today, we do not provide collection parameters, so the comment is not quite accurate I think. Here is what I propose instead:
```suggestion
	 * Retrieves all languages.
```
Also I wonder what happens for deactivated languages? And if we should show them or  not here.

Reply:
To be discussed with the team.